mmv1/third_party/terraform/resources/resource_storage_transfer_job.go

```diff
@@ -349,7 +349,7 @@ func awsS3DataSchema() *schema.Resource {
 			},
 			"aws_access_key": {
 				Type:     schema.TypeList,
-				Required: true,
+				Optional: true,
```

Comment:
Now that we are moving this from Required -> Optional can we add an AtLeastOneOf with `aws_access_key` & `role_arn` so that one of them must be specified?

We try to avoid allowing empty blocks in configs as they tend to cause problems

Reply:
+1 , either of those is required, won't work with neither

Reply:
Does it work with both set? If not, we can use ExactlyOneOf instead

Reply:
Technically the backend allows you to set both but

* There's no reason to do so
* aws_access_key is never returned by the service
* If role_arn is set it takes priority

so I think it's fine to wrap this in ExactlyOneOf even though the underlying proto couldn't wrap these fields in a OneOf to maintain backwards compatibility.